activerecord/test/cases/calculations_test.rb

```diff
@@ -460,6 +460,44 @@ def test_should_count_manual_select_with_include
     assert_equal 6, Account.select("DISTINCT accounts.id").includes(:firm).count
   end
 
+  def test_should_count_manual_select_with_count_all
+    assert_equal 5, Account.select("DISTINCT accounts.firm_id").count(:all)
+  end
+
+  def test_should_count_with_manual_distinct_select_and_distinct
+    assert_equal 4, Account.select("DISTINCT accounts.firm_id").distinct(:id).count
+  end
+
+  def test_should_count_manual_select_with_group_with_count_all
+    c = Account.select("DISTINCT accounts.firm_id").group("accounts.firm_id").count(:all)
+    assert_equal([[nil, 1], [1, 1], [2, 1], [6, 2], [9, 1]].to_h, c)
```

Comment:
Don't use `.to_h`, but just a hash like here.

https://github.com/rails/rails/blob/1cf72affe0ac0c7dac557fe463d2153c758bf7af/activerecord/test/cases/calculations_test.rb#L765-L769

Reply:
I tried to use:
`{ nil => 1, 1 => 1, 2 => 1, 6 => 2, 9 => 1 }` 
But code climate threw errors